### PR TITLE
Upgrade android compileSdkVersion 34

### PR DESCRIPTION
**Pull request by AlexandraDamaschin**
We should keep things updated in the android project, created this PR in order to update them as we need this library and isn't complied anymore with the other requirements. 

I've updated: 
* kotlin_version = '1.8.22'
* compileSdkVersion 34

Types of changes
What types of changes does your code introduce? Put an x in all the boxes that apply:

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

Checklist
Go over all the following points, and put an x in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask.

- [X]  My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X]  I have updated the documentation accordingly.

### CODE REVIEW

**Comment by AlexandraDamaschin** [comment]
@NMGuner maybe you could help here? We would need this fix deployed soon. Thanks in advance

**Comment by MasoudFallahpourbaee** [comment]
Hi @AlexandraDamaschin,
Can you please tell us a little bit more about "isn't complied anymore with the other requirements." In other words, what error you get when you use `klarna-mobile-sdk-flutter` as is?

**Comment by AlexandraDamaschin** [comment]
> Hi @AlexandraDamaschin, Can you please tell us a little bit more about "isn't complied anymore with the other requirements." In other words, what error do you get when you using `klarna-mobile-sdk-flutter` as is?

@MasoudFallahpourbaee sure, this is the error I get using klarna_mobile_sdk_flutter: ^1.0.1

```
Execution failed for task ':klarna_mobile_sdk_flutter:compileDebugKotlin'.
> 'compileDebugJavaWithJavac' task (current target is 1.8) and 'compileDebugKotlin' task (current target is 17) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain
```

**Comment by AlexandraDamaschin** [comment]
> @AlexandraDamaschin Can you please sign the CLA. Once done, I'll merge the PR and release a new version.

I just signed the CLA.  Looking forward for the new upgrade.

File: android/build.gradle [hunk 1]
```diff
@@ -44,6 +44,12 @@ android {
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
+    }
+        kotlinOptions {
```
Comment (MasoudFallahpourbaee) [inline android/build.gradle:48]: The indentation is not correct here. Could you fix that please?

Comment (AlexandraDamaschin) [reply android/build.gradle]: Done

**Review by MasoudFallahpourbaee — APPROVED** [review]

**Comment by MasoudFallahpourbaee** [comment]
Thanks!

**Pull request merged**